R/deployApp.R

```diff
@@ -432,7 +432,7 @@ deployApp <- function(appDir = getwd(),
     taskComplete(quiet, "Visibility updated")
   }
   if (length(target$envVars) > 0) {
-    taskStart(quiet, "Updating environment variables {envVars}...")
+    taskStart(quiet, "Updating {length(envVars)} environment variable{?s}: {.field {names(envVars)}}...")
```

Comment:
These are already just the names of the env vars?

Reply:
Sorry, I had a hard time following the flow for that property. What happened for me -- before I applied the other fix -- was that I tried to deploy an app with one env var to shinyapps.io. Something like `envVars = c("FLAG" = "true")`. This made it past the first check and failed here (because the shinyapps `client` doesn't implement `$setEnvVars()`) at which point `envVars` was `"true"`.

I made this change first and then realized that the earlier check was failing, and in my probably semi-broken state `envVars` was a named vector. This may have been the result of a local rsconnect deploy record with a value that shouldn't have been written?

Reply:
Yeah, maybe we need some earlier validation that `envVars` is an unnamed character vector?

Reply:
Ohhh, I read the documentation more carefully and realized that the environment variables should be set in the current session and the user should provide just the names, as in `envVars = c("FLAG", "OTHER_FLAG")`.

I think my confusion stemmed from being familiar with connectapi, where there's an interface for setting environment variables during deployment that uses the `FLAG = "value"` pattern.

I updated the docs to give a little guidance about how those values should be set, and I added a check that the vector provided to `envVars` is unnamed. This feels a little aggressive but I think it's worth it, given the potential risk of leaking a credential if a user misunderstands how `envVars` should be used.

Reply:
In case it's not clear, the interface looks like this is to prevent you from typing the values of env vars in executed code since that will be captured in `.Rhistory`.